azavea/district-builder-dtl-pa
# Add plan submission fields to user profile

## Overview
In the plan submission, some fields we require are common to the user and make sense to store for the user and pre-populate as a convenience to the user. The fields are:
- County
- Contest division
- Organization/school

This PR adds these fields to the Profile model (Except organization, which was already present), requires them to be entered for new user registration, and includes a migration that will populate the fields from the user's most recent submission (If the user has one). Users that already exist but without any submissions will have blank values for these fields.

### Checklist


### Demo
![screen shot 2019-01-21 at 4 16 34 pm](https://user-images.githubusercontent.com/1032849/51499350-f0e02780-1d97-11e9-8259-d96094bff32b.png)

![screen shot 2019-01-21 at 4 16 14 pm](https://user-images.githubusercontent.com/1032849/51499359-f5a4db80-1d97-11e9-871d-77977cbb68af.png)

![screen shot 2019-01-22 at 2 43 32 pm](https://user-images.githubusercontent.com/1032849/51561058-4548dd00-1e54-11e9-8a3f-06323dc57eff.png)


### Notes
- For user accounts that already exist but have no submissions, their submissions will not have any value pre-populated and this will not be automatically rectified unless the user edits their account and sets the values there. This is in line with the existing behavior for First Name/Last Name, which is stored on the submission as entered but the change is not reflected on the user's account unless done there explicitly.

## Testing Instructions
- Prior to installing migrations and on `develop`...
  - Create a user with minimal configuration
  - Create a second user and submit a plan for that user with Organization, County and Division specified
- Switch to branch and install migrations
  - Migrations should run successfully
- Sign in to user with submitted plan
- Open "My Account"
  - Organization, County and Division should be shown, and have values of the plan submitted previously
- Create a new plan and attempt to submit
  - Fields Organization, County and Division should have values of the plan submitted previously
- Sign in to user with no submitted plan and minimal configuration
- Open "My Account"
  - Organization, County and Division should be shown and should _*not*_ have any values
- Create a plan and attempt to submit
  - Relevant fields should be blank, but plan submit successfully when filled
- Attempt to register a new account, but leave Organization, County and Division as default
  - Registration should fail
- Fill in fields and attempt to register again
  - Registration should succeed

Closes [PT162929726](https://www.pivotaltracker.com/story/show/162929726)
